src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -266,6 +266,47 @@ public bool IsSolutionAvailable
             }
         }
 
+        public bool IsSolutionDPLEnabled
+        {
+            get
+            {
+#if VS15
+                return ThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                    var vsSolution7 = _vsSolution as IVsSolution7;
+
+                    if (vsSolution7 != null && vsSolution7.IsSolutionLoadDeferred())
+                    {
+                        return true;
+                    }
+
+                    return false;
+                });
+#elif VS14
```

Comment:
For VS16 we'd need to revisit this code.  This would be more forward-compatible by default:

```
#if VS14
#else
#endif
```